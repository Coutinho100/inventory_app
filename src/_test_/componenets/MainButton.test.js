import React from 'react';
import MainButton from '../../components/Component/MainButton';
import { View, Text, TouchableOpacity } from 'react-native';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


describe("<Button view/>", () => {
    test('renders correctly', () => {
        const tree = renderer.create(<MainButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders the length of view', () => {
        const wrapper = shallow(<MainButton />);
        expect(wrapper.find(View).length).toEqual(1);
        expect(wrapper.find(TouchableOpacity).length).toEqual(1);
        expect(wrapper.find(Text).length).toEqual(1);
    })

});