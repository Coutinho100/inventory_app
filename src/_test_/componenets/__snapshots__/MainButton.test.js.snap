// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button view/> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#cccccc",
        "borderRadius": 25,
        "margin": 3,
        "paddingHorizontal": 15,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "open-sans",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
